src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -991,7 +1007,8 @@ RCLConsensus::Adaptor::preStartRound(
 {
     // We have a key, we do not want out of sync validations after a restart
     // and are not amendment blocked.
-    validating_ = validatorKeys_.publicKey.size() != 0 &&
+    validating_ = validatorKeys_.keys.has_value() &&
+        validatorKeys_.keys->publicKey != PublicKey::emptyPubKey &&
```

Comment:
It should be impossible for `keys` to be initialized with `PublicKey::emptyKey`. The check is still good, but I suggest adding an assert before this line:
```
assert(!validatorKeys_.keys || validatorKeys_.keys->publicKey != PublicKey::emptyPubKey);
```
Also there's not much need for `has_value()` unless you're dealing with ambiguity.